File: Family tree/src/ui/Menu.java
package ui;

import java.io.UnsupportedEncodingException;
import java.util.ArrayList;
import java.util.List;

import ui.commands.AddHuman;
import ui.commands.Finish;
import ui.commands.GetFamilyTree;

public class Menu {
    List<Command> menu;

    public Menu (Console console){
        menu = new ArrayList<>();
        menu.add(new AddHuman(console));
        menu.add(new GetFamilyTree(console));
        menu.add(new Finish(console));
    }


    public String print(){
        StringBuffer sb = new StringBuffer();
        for (int i = 0; i < menu.size(); i++) {
            sb.append(i+1 + ". " + menu.get(i).getDescription()+"\n");
        }
        return sb.toString();
    }

    public int getSize(){
        return menu.size();
    }

    void execute(int choice){
        menu.get(choice-1).execute();
    }
}

File: ui/menu/Menu.cs
using System.Text;

public class Menu
{
    private List<Command> commandList;


    public Menu(OurConsole ourConsole)
    {
        commandList = new List<Command>();
        commandList.Add(new AddGod(ourConsole));
        commandList.Add(new AddGods(ourConsole));
        commandList.Add(new GetFamilyTreeInfo(ourConsole));
        commandList.Add(new Finish(ourConsole));
    }

    public string print()
    {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < commandList.Count; i++)
        {
            sb.Append($"{i + 1}. {commandList[i].getDescription()}\n");
        }
        return sb.ToString();
    }

    public void execute(String choice)
    {
        commandList[Convert.ToInt32(choice) - 1].execute();
    }
}